src/varlinkctl/varlinkctl.c

```diff
@@ -289,56 +292,129 @@ typedef struct GetInterfaceDescriptionData {
+        }
+
+        /* Automatically switch on JSON_SEQ if we output multiple JSON objects */
+        if (!list_methods && strv_length(interfaces) > 1)
+                arg_json_format_flags |= JSON_FORMAT_SEQ;
+
+        _cleanup_strv_free_ char **methods = NULL;
+
+        STRV_FOREACH(i, interfaces) {
+                JsonVariant *reply = NULL;
+                r = varlink_callb_and_log(
+                                vl,
+                                "org.varlink.service.GetInterfaceDescription",
+                                &reply,
+                                JSON_BUILD_OBJECT(JSON_BUILD_PAIR_STRING("interface", *i)));
                 if (r < 0)
                         return r;
 
-                /* Try to parse the returned description, so that we can add syntax highlighting */
-                r = varlink_idl_parse(ASSERT_PTR(description), &line, &column, &vi);
-                if (r < 0) {
-                        log_warning_errno(r, "Failed to parse returned interface description at %u:%u, showing raw interface description: %m", line, column);
+                if (FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF) || list_methods) {
+                        static const struct JsonDispatch dispatch_table[] = {
+                                { "description",  JSON_VARIANT_STRING, json_dispatch_const_string, 0, JSON_MANDATORY },
+                                {}
+                        };
+                        _cleanup_(varlink_interface_freep) VarlinkInterface *vi = NULL;
+                        const char *description = NULL;
+                        unsigned line = 0, column = 0;
 
-                        fputs(description, stdout);
-                        if (!endswith(description, "\n"))
-                                fputs("\n", stdout);
+                        r = json_dispatch(reply, dispatch_table, JSON_LOG|JSON_ALLOW_EXTENSIONS, &description);
+                        if (r < 0)
+                                return r;
+
+                        if (!list_methods && i > interfaces)
+                                print_separator();
+
+                        /* Try to parse the returned description, so that we can add syntax highlighting */
+                        r = varlink_idl_parse(ASSERT_PTR(description), &line, &column, &vi);
+                        if (r < 0) {
+                                if (list_methods)
+                                        return log_error_errno(r, "Failed to parse returned interface description at %u:%u: %m", line, column);
+
+                                log_warning_errno(r, "Failed to parse returned interface description at %u:%u, showing raw interface description: %m", line, column);
+
+                                pager_open(arg_pager_flags);
+                                fputs(description, stdout);
+                                if (!endswith(description, "\n"))
+                                        fputs("\n", stdout);
```

Comment:
Hmm, I prefer `puts(delete_trailing_chars(description, NULL))`, but doesn't matter

Reply:
This would mean we'd have to copy the string first. Right now `description` string is a const pointer directly int the json object...

Reply:
I went down another route now, and added a helper fputs_with_newline() since we have the same pattern elsewhere. It works with const strings, but hides the conditioning in the helper. I think this is a nicer solution.